### PR TITLE
[iCubClient,ARE] add method to pointfar

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
This adds the ability to call the recently added pointfar feature of ARE (see https://github.com/robotology/icub-main/pull/413). Thanks @pattacini for providing this method.
I haven't tested this yet on the robot. We will do tomorrow once @hjchang is back in the lab.

Best,
Tobi

### CODE REVIEW

File: main/src/libraries/wrdac/src/subsystems/subSystem_ARE.cpp [hunk 1]
```diff
@@ -325,6 +325,52 @@ bool wysiwyd::wrdac::SubSystem_ARE::push(const yarp::sig::Vector &targetUnsafe,
...
+    yarp::os::Bottle bCmd;
+    bCmd.addVocab(yarp::os::Vocab::encode("pfar"));
+
+    yarp::sig::Vector target=targetUnsafe;
+    yarp::os::Bottle opt=options;
+    selectHandCorrectTarget(opt,target);
+    target=applySafetyMargins(target);
+    appendCartesianTarget(bCmd,target);
+    bCmd.append(opt);
+
```
Comment (pattacini) [inline main/src/libraries/wrdac/src/subsystems/subSystem_ARE.cpp]: Better to do:
```cpp
yarp::sig::Vector target=applySafetyMargins(targetUnsafe);
appendCartesianTarget(bCmd,target);
bCmd.append(opt);
```
We don't have to apply IOL corrections to far away targets. Then, the hand is selected internally by `ARE`.

**Comment by Tobias-Fischer** [comment]
Good point @pattacini. I changed it.

File: main/src/libraries/wrdac/src/subsystems/subSystem_ARE.cpp [hunk 2]
```diff
@@ -343,10 +343,7 @@ bool wysiwyd::wrdac::SubSystem_ARE::pointfar(const yarp::sig::Vector &targetUnsa
     yarp::os::Bottle bCmd;
     bCmd.addVocab(yarp::os::Vocab::encode("pfar"));
 
-    yarp::sig::Vector target=targetUnsafe;
-    yarp::os::Bottle opt=options;
-    selectHandCorrectTarget(opt,target);
-    target=applySafetyMargins(target);
+    yarp::sig::Vector target=applySafetyMargins(target);
```
Comment (pattacini) [inline main/src/libraries/wrdac/src/subsystems/subSystem_ARE.cpp]: It should be rather:
```cpp
yarp::sig::Vector target=applySafetyMargins(targetUnsafe);
```

Comment (Tobias-Fischer) [inline main/src/libraries/wrdac/src/subsystems/subSystem_ARE.cpp]: Good catch, thanks.

**Comment by Tobias-Fischer** [comment]
Hi @pattacini, thanks for the catch. We are likely going to test this today - do you think we can merge?

Best,
Tobi

**Comment by pattacini** [comment]
I think so 👍

**Pull request merged**